app/src/main/java/com/bogdan/codeforceswatcher/features/MainActivity.kt

```diff
@@ -120,10 +121,27 @@ class MainActivity : AppCompatActivity(), StoreSubscriber<UIState> {
 
     private fun onActionsTabSelected() {
         llSorting.visibility = View.GONE
-        fab.setOnClickListener(null)
+        fab.setOnClickListener { showShareDialog() }
         fab.setImageDrawable(getDrawable(R.drawable.ic_share))
     }
 
+    private fun showShareDialog() {
+        AlertDialog.Builder(this)
+            .setTitle(getString(R.string.share_cw))
+            .setMessage(getString(R.string.help_cw_make_more_social))
+            .setCancelable(false)
+            .setPositiveButton(getString(R.string.share)) { _, _ -> share() }
+            .setNegativeButton(getString(R.string.cancel), null)
+            .create()
+            .show()
+    }
+
+    private fun share() = Intent(Intent.ACTION_SEND).apply {
```

Comment:
It's little but misleading here as can be read like `share an Intent`. Instead you could probably write it like this:
```
private fun share() = startActivity(Intent().apply {
    action = Intent.ACTION_SEND
    type = "text/plain"
    putExtra(Intent.EXTRA_TEXT, getString(R.string.share_cw_message))
})
```